GVRf/Framework/framework/src/main/java/org/gearvrf/GVRSceneObject.java

```diff
@@ -699,11 +699,20 @@ public GVRSceneObject getParent() {
      * @param child
      *            {@link GVRSceneObject Object} to add as a child of this
      *            object.
+     * @return true if child was added, else false
      */
-    public void addChildObject(GVRSceneObject child) {
+    public boolean addChildObject(GVRSceneObject child) {
+        if (child.mParent == this) {
+            return false;
+        }
+        if (child.mParent != null) {
+            Log.e("gvrf", "Cannot add child - it has another parent");
```

Comment:
I did that and it broke immersivepedia. I agree that it should throw an exception. I figured out how to fix immersivepedia too so I will do that also.